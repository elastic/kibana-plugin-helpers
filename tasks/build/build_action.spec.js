/*eslint-env jest*/
const resolve = require('path').resolve;
const fs = require('fs');

const del = require('del');

const buildAction = require('./build_action');

const PLUGIN_FIXTURE = resolve(__dirname, '__fixtures__/test_plugin');
const PLUGIN_BUILD_DIR = resolve(PLUGIN_FIXTURE, 'build');
const PLUGIN = require('../../lib/plugin_config')(PLUGIN_FIXTURE);

describe('build_action', () => {
  beforeEach(() => del(PLUGIN_BUILD_DIR));
  afterEach(() => del(PLUGIN_BUILD_DIR));

  it('creates a zip in the build directory', () => {
    return buildAction(PLUGIN).then(() => {
      if (!fs.existsSync(resolve(PLUGIN_BUILD_DIR, PLUGIN.id + '-' + PLUGIN.version + '.zip'))) {
        throw new Error('expected the plugin to build a zip file');
      }
    });
  });
});