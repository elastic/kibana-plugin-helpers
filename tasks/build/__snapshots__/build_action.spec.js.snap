// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build_action calling create_build rejects returned promise when build fails 1`] = `"foo bar"`;
